cdap-docs/build.sh

```diff
@@ -38,321 +72,379 @@ function set_project_path() {
+  echo "--------------------------------------------------------"
+  ring_bell "Completed \"build all\""
+  echo "========================================================"
+  echo "========================================================"
+  echo
+  exit ${warnings}
+}
 
-  # Build outer-level docs
-  cd ${SCRIPT_PATH}
-  cp ${COMMON_HIGHLEVEL_PY}  ${BUILD}/${SOURCE}/conf.py
-  cp -R ${COMMON_IMAGES}     ${BUILD}/${SOURCE}/
-  cp ${COMMON_SOURCE}/*.rst  ${BUILD}/${SOURCE}/
-  
-  if [ "x${1}" == "x" ]; then
-    sphinx-build -b html -d build/doctrees build/source build/html
-  else
-    sphinx-build -D googleanalytics_id=${1} -D googleanalytics_enabled=1 -b html -d build/doctrees build/source build/html
+function build_docs() {
+  local warnings
+  local doc_type=${1}
+  local source_path=${ARG_2}
+  local javadocs="${WITHOUT}"
+  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${WEB}" ]; then
+    javadocs="${WITH}"
   fi
-}
-  
-function copy_docs_lower_level() {
-  echo ""
   echo "========================================================"
-  echo "Copying lower-level documentation..."
+  echo "========================================================"
+  echo "Building \"${doc_type}\" (${javadocs} Javadocs)"
   echo "--------------------------------------------------------"
-  echo ""
-
-  for i in ${MANUALS}; do
-    copy_html ${i}
-  done
-
-  local project_dir
-  # Rewrite 404 file, using branch if not a release
-  if [ "x${GIT_BRANCH_TYPE}" == "xfeature" ]; then
-    project_dir=${PROJECT_VERSION}-${GIT_BRANCH}
+  echo
+  if [ "${doc_type}" != "${DOCS}" ]; then
+    clean_targets
+  fi
+  clear_messages_set_messages_file
+  run_command docs-first-pass ${source_path}
+  if [ "${doc_type}" == "${DOCS}" ]; then
+    build_docs_outer_level ${source_path}
+    copy_docs_inner_level
   else
-    project_dir=${PROJECT_VERSION}
+    clear_messages_set_messages_file
   fi
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_static"  "src=\"/cdap/${project_dir}/en/_static"
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_images"  "src=\"/cdap/${project_dir}/en/_images"
-  rewrite ${BUILD}/${HTML}/404.html "/href=\"http/!s|href=\"|href=\"/cdap/${project_dir}/en/|g"
-  rewrite ${BUILD}/${HTML}/404.html "action=\"search.html"  "action=\"/cdap/${project_dir}/en/search.html"
-}
-
-################################################## current
-
-function build_all() {
-  echo ""
-  echo "========================================================"
-  echo "========================================================"
-  echo "Building All: GitHub and Web Docs."
+  if [ "${javadocs}" == "${WITH}" ]; then
+    run_command javadocs
+  fi
+  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${GITHUB_ONLY}" ]; then
+    run_command docs-github-part ${source_path}
+  elif [ "${doc_type}" == "${WEB}" -o "${doc_type}" == "${WEB_ONLY}" ]; then
+    run_command docs-web-part ${source_path}
+  fi
+  display_version
+  display_messages_file
+  warnings=$?
```

Comment:
Yes...``display_messages_file`` determines the return code based on the contents of the message file. Once all the messages have been displayed at what will become the end of the log, it then returns the code and Bamboo can decide what to do with the results. This way, when you go to the Bamboo page, you should be able to see at a glance that there are error messages to be reviewed, and which manuals to poke into.